cmd/appengine/device.go

```diff
@@ -1276,6 +1276,21 @@ func devicesPublishDataStreamF(command *cobra.Command, args []string) error {
 					}
 					aggrPayload[k] = acc
 				}
+				if payloadType == interfaces.IntegerArray {
+					acc := make([]int32, 0)
+
+					for _, item := range val {
+						acc = append(acc, int32(item.(float64)))
+					}
+					aggrPayload[k] = acc
+				}
+				if payloadType == interfaces.LongIntegerArray {
+					acc := make([]int64, 0)
+					for _, item := range val {
+						acc = append(acc, int64(item.(float64)))
+					}
+					aggrPayload[k] = acc
+				}
```

Comment:
See my previous comment: these lines are inside `devicesPublishDataStreamF` but not in `devicesSetPropertyF`